# Ex1 - Getting and Knowing your Data



### Step 1. Import the necessary libraries

In [2]:
import numpy as np
import pandas as pd

### Step 2. Import the dataset

In [3]:
pd.read_csv("D:/Ad_py_lib.csv/pandas_and_numpy/pandas_data.csv")

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

### Step 3. Assign it to a variable called a.

In [ ]:
a = pd.read_csv("D:/Ad_py_lib.csv/pandas_and_numpy/pandas_data.csv")
a

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [ ]:
a = pd.read_csv("D:/Ad_py_lib.csv/pandas_and_numpy/pandas_data.csv",index_col = "order_id")
a

quantity                              item_name  \
order_id                                                    
1                1           Chips and Fresh Tomato Salsa   
1                1                                   Izze   
1                1                       Nantucket Nectar   
1                1  Chips and Tomatillo-Green Chili Salsa   
2                2                           Chicken Bowl   
...            ...                                    ...   
1833             1                          Steak Burrito   
1833             1                          Steak Burrito   
1834             1                     Chicken Salad Bowl   
1834             1                     Chicken Salad Bowl   
1834             1                     Chicken Salad Bowl   

                                         choice_description item_price  
order_id                                                                
1                                                       NaN     $2.39   
1                                              [Clementine]     $3.39   
1                                                   [Apple]     $3.39   
1                                                       NaN     $2.39   
2         [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                     ...        ...  
1833      [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
1833      [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
1834      [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
1834      [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
1834      [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 4 columns]

### Step 4. See the first 10 entries

In [ ]:
a.head(10)

quantity                              item_name  \
order_id                                                    
1                1           Chips and Fresh Tomato Salsa   
1                1                                   Izze   
1                1                       Nantucket Nectar   
1                1  Chips and Tomatillo-Green Chili Salsa   
2                2                           Chicken Bowl   
3                1                           Chicken Bowl   
3                1                          Side of Chips   
4                1                          Steak Burrito   
4                1                       Steak Soft Tacos   
5                1                          Steak Burrito   

                                         choice_description item_price  
order_id                                                                
1                                                       NaN     $2.39   
1                                              [Clementine]     $3.39   
1                                                   [Apple]     $3.39   
1                                                       NaN     $2.39   
2         [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
3         [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
3                                                       NaN     $1.69   
4         [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
4         [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
5         [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

### Step 5. What is the number of observations in the dataset?

In [11]:
# a.shape[0] is simply the total number of rows in your DataFrame.
# You are asking for the first element of the tuple that a.shape returns.
a.shape[0]

4622

In [18]:
# a.shape gives you the dimensions of the DataFrame as a tuple (rows, columns).
a.shape

(4622, 4)

In [20]:
# a.dtypes

# a.tail(15)

a.size

18488

In [ ]:
# show you the difference between a.shape[0] vs len(a) (they look similar but can differ in some cases)?


### Step 6. What is the number of columns in the dataset?

In [ ]:
a.shape[1]

4

### Step 7. Print the name of all the columns.

In [ ]:
a.columns

Index(['quantity', 'item_name', 'choice_description', 'item_price'], dtype='object')

### Step 8. How is the dataset indexed?

In [1]:
a.index

NameError: name 'a' is not defined

### Step 9. Which was the most-ordered item?

In [26]:
# If no quantity column → use a.value_counts().idxmax()
a.value_counts().idxmax() #this is wrong for this dataframe

(np.int64(1), 'Canned Soft Drink', '[Coke]', '$1.25 ')

In [25]:
a.groupby("item_name")["quantity"].sum().idxmax()

'Chicken Bowl'

In [24]:
a.groupby('item_name')["quantity"].sum().sort_values(ascending=False).head(1)

item_name
Chicken Bowl    761
Name: quantity, dtype: int64

* result.max() → 761 (the highest quantity)

* result.idxmax() → "Chicken Bowl" (the item with that highest quantity)

Summary:

* If no quantity column → use .value_counts().idxmax()

* If quantity exists → use .groupby('item')['quantity'].sum().idxmax()

### Would you like me to also show you the same for **top 3 most-ordered items** (instead of only the max one)?

In [33]:
top3= a.groupby("item_name")["quantity"].sum().sort_values(ascending=False).head(3)
print(top3)

item_name
Chicken Bowl           761
Chicken Burrito        591
Chips and Guacamole    506
Name: quantity, dtype: int64


### Step 10. For the most-ordered item, how many items were ordered?

### Step 11. What was the most ordered item in the choice_description column?

### Step 12. How many items were orderd in total?

### Step 13. Turn the item price into a float

#### Step 13.a. Check the item price type

#### Step 13.b. Create a lambda function and change the type of item price

#### Step 13.c. Check the item price type

### Step 14. How much was the revenue for the period in the dataset?

### Step 15. How many orders were made in the period?

### Step 16. What is the average revenue amount per order?

### Step 17. How many different items are sold?